cmd/nvidia-dra-controller/imex.go

```diff
@@ -152,7 +164,12 @@ func (d DriverResources) DeepCopy() resourceslice.DriverResources {
 		Pools: make(map[string]resourceslice.Pool),
 	}
 	for p := range d.Pools {
-		driverResources.Pools[p] = generateImexChannelPool(p, ImexChannelLimit)
+		id := strings.Split(p, ".")
+		clique, err := strconv.Atoi(id[1])
+		if err != nil {
+			klog.Errorf("Error converting string to int: %v", err)
+		}
+		driverResources.Pools[p] = generateImexChannelPool(id[0], clique)
```

Comment:
We should probably just update upstream to generate a DeepCopy function for `DriverResources` instead of trying to roll our own. Then none of this is necessary.

Reply:
https://github.com/kubernetes/kubernetes/pull/128331

Reply:
Merged. This PR now has a patched version of that PR.